src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -11326,6 +11853,106 @@ static bool TakeOutParam<T>(T y, out T x)
         }
 
         [Fact]
+        [CompilerTrait(CompilerFeature.Tuples)]
+        public void Scope_SwitchLabelGuard_13()
+        {
+            var source =
+@"
+public class X
+{
+    public static void Main()
+    {
+        Test(1);
+    }
+
+    static void Test(int val)
+    {
+        switch (val)
+        {
+            case 1 when TakeOutParam(123, out var x1):
+                var (z0, z1)  = (x1 > 0, TakeOutParam(x1, out var y1));
+                System.Console.WriteLine(y1);
+                System.Console.WriteLine(z1 ? 1 : 0);
+                break;
+        }
+    }
+
+    static bool TakeOutParam<T>(T y, out T x) 
+    {
+        x = y;
+        return true;
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, 
+                                                              options: TestOptions.DebugExe, parseOptions: TestOptions.Regular);
+
+            CompileAndVerify(compilation, expectedOutput:
+@"123
+1").VerifyDiagnostics();
+
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var yRef = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "y1").Single();
```

Comment:
Changed throughout the file.

---
In reply to: [76287603](https://github.com/dotnet/roslyn/pull/13344#discussion_r76287603) [](ancestors = 76287603)